engine/src/test/java/org/hibernate/validator/test/internal/engine/valuehandling/UnwrappingTest.java

```diff
@@ -138,84 +139,107 @@ public void validate_wrapped_value_if_value_extractor_unwraps_by_default_and_unw
 		assertCorrectConstraintTypes( constraintViolations, Min.class );
 	}
 
-	public class Foo {
+	@Test(expectedExceptions = ConstraintDeclarationException.class, expectedExceptionsMessageRegExp = "HV000199.*")
```

Comment:
It'd make sense to have this and others in the TCK eventually. But it's good here for now, I've made it a talking point for next week to discuss the TCK.